mycelium/src/tun/darwin.rs

```diff
@@ -174,6 +172,53 @@ fn validate_utun_name(input: &str) -> bool {
         .is_ok()
 }
 
+/// Find an available utun interface name
+fn find_available_utun_name(preferred_name: &str) -> Result<String, io::Error> {
+    let interfaces = datalink::interfaces();
+    let utun_interfaces: Vec<_> = interfaces
```

Comment:
how about put this line after the `if !preferred_name.is_empty() ` block?

Reply:
Indeed, would be uncessary in case `preferred_name` is valid. Thanks!